src/pkg/services/m365/api/pagers/pagers_test.go

```diff
@@ -305,300 +305,588 @@ func (suite *PagerUnitSuite) TestBatchEnumerateItems() {
 	}
 }
 
-func (suite *PagerUnitSuite) TestGetAddedAndRemovedItemIDs() {
-	type expected struct {
-		added         []testItem
-		removed       []string
-		deltaUpdate   DeltaUpdate
-		validModTimes bool
+func assertSliceEmptyOr[S ~[]E, E any](
```

Comment:
nit: I like `T` as the standard "it's the data type".  I assume `E` stands for `Element`?
```suggestion
func assertSliceEmptyOr[S ~[]T, T any](
```